bc_obps/registration/migrations/0052_auto_20241101_1845.py

```diff
@@ -0,0 +1,43 @@
+# Generated by Django 5.0.7 on 2024-11-01 18:45
+
+from django.db import migrations
+
+# Generated by Django 5.0.7 on 2024-11-01 18:38
+
+
+def migrate_to_bc_greenhouse_gas_id_model(apps, schema_monitor):
+    '''
+    Create BcGreenhouseGasId records for each operation and facility that have a bcghg_id field (they will be migrated to a OneToOneField that references the BcGreenhouseGasId model in the next migration).
+
+    '''
+    Operation = apps.get_model('registration', 'Operation')
+    Facility = apps.get_model('registration', 'Facility')
+    BcGreenhouseGasId = apps.get_model('registration', 'BcGreenhouseGasId')
+
+    # operation
+    operations_with_bcghg_id = Operation.objects.filter(bcghg_id__isnull=False)
+    for operation in operations_with_bcghg_id:
+        BcGreenhouseGasId.objects.create(
+            id=operation.bcghg_id,
+        )
+
+    # facility
+    facilities_with_bcghg_id = Facility.objects.filter(bcghg_id__isnull=False)
+
+    for facility in facilities_with_bcghg_id:
+        BcGreenhouseGasId.objects.create(
+            id=facility.bcghg_id,
+        )
+
+    # check counts
+    if BcGreenhouseGasId.objects.count() != operations_with_bcghg_id.count() + facilities_with_bcghg_id.count():
+        raise Exception('Migration unsuccessful')
```

Comment:
FYI: Raising exceptions works great. I have tested that `assert BcGreenhouseGasId.objects.count() == operations_with_bcghg_id.count()`, and it also worked.